paddle/fluid/eager/tensor_wrapper.h

```diff
@@ -70,24 +70,25 @@ class TensorWrapper {
         }
       }
       return;
-    }
-
-    // shallow copy tensor_impl here
-    if (no_need_buffer) {
-      if (phi::DenseTensor::classof(tensor.impl().get())) {
-        // Only Copy Meta
-        phi::DenseTensor* dense_tensor =
-            static_cast<phi::DenseTensor*>(tensor.impl().get());
-        auto tw_dense_tensor = std::make_shared<phi::DenseTensor>();
-        tw_dense_tensor->set_meta(dense_tensor->meta());
-        intermidiate_tensor_.set_impl(tw_dense_tensor);
+    } else {
```

Comment:
since first part of if will return, it seems no need to build a extra else branch

Reply:
done, revert this branch, thx